roles/matrix-nginx-proxy/defaults/main.yml

```diff
@@ -646,3 +646,9 @@ matrix_nginx_proxy_synapse_cache_proxy_cache_valid_time: "24h"
 # http://nginx.org/en/docs/ngx_core_module.html#worker_connections
 matrix_nginx_proxy_worker_processes: auto
 matrix_nginx_proxy_worker_connections: 1024
+
+# A mapping of JVB ids to hostname for additional jvb blocks to add to Jitsi's server configuration (matrix-jitsi.conf)
+matrix_nginx_proxy_proxy_jitsi_additional_jvbs_enabled: false
+matrix_nginx_proxy_proxy_jitsi_additional_jvbs:
+  - jvb-2: 192.168.0.1
+  - jvb-3: 192.168.0.2
```

Comment:
Using a default value which doesn't contain stuff would be better.

```suggestion
# Example:
# matrix_nginx_proxy_proxy_jitsi_additional_jvbs:
#   jvb-2: 192.168.0.1
#   jvb-3: 192.168.0.2
matrix_nginx_proxy_proxy_jitsi_additional_jvbs: {}
```

Doing this, you don't even need the `_enabled` variable.

You can just iterate over it in `roles/matrix-nginx-proxy/templates/nginx/conf.d/matrix-jitsi.conf.j2`. If `matrix_nginx_proxy_proxy_jitsi_additional_jvbs` is empty, nothing happens. If it's non-empty (JVBs added to the list), it will inject a `location` block for each one.

My `matrix_nginx_proxy_proxy_jitsi_additional_jvbs` above uses a dictionary, which can be iterated over directly like this: `{% for id, ip_address in matrix_nginx_proxy_proxy_jitsi_additional_jvbs %}`, which removes the need for 2 nested `for` loops.

----

I'm also a bit confused about how `matrix_nginx_proxy_proxy_jitsi_additional_jvbs` is meant to be used. Maybe we need more comments there and in `docs/configuring-playbook-jitsi.md`.

Can `matrix_nginx_proxy_proxy_jitsi_additional_jvbs` be populated automatically based on your inventory? Even if it can, should it?

Do we still need `matrix_jitsi_jvb_server_id: jvb-1`?

It seems like using `jvb-1` inside `matrix_nginx_proxy_proxy_jitsi_additional_jvbs` will cause trouble. Perhaps this should be pointed out. Maybe pointing to `matrix_jitsi_jvb_server_id` might be a good idea.

`matrix_jitsi_jvb_server_id`'s comment (if this variable will remain in use) could also point to `matrix_nginx_proxy_proxy_jitsi_additional_jvbs`.

I'm still not sure how it all works. How do you tell Jitsi to use these other JVBs? The reverse proxy config is there, but what tells Jitsi that there's `jvb-2` as well? Is this part still missing from the PR or am I misunderstanding things?

Reply:
Yeah sorry, I wanted to test that it all worked before adding some more documentation about how to use. 

We will need the matrix_jitsi_jvb_server_id because the JVB themselves need to know their server id. This field ends up in JVB_WS_SERVER_ID environment variable of the additional JVB once it has been provisioned.